src/rust/engine/process_executor/src/test_server.rs

```diff
@@ -0,0 +1,159 @@
+use std::fmt::Debug;
+use std::ops::Deref;
+use std::slice::Iter;
+use std::sync::{Arc, Mutex};
+
+use bazel_protos;
+use grpcio;
+use protobuf;
+
+#[derive(Clone, Debug)]
+pub struct MockExecution {
+  ///
+  /// The name of the operation. It is assumed that all operation_responses use this name.
+  ///
+  pub name: String,
+  ///
+  /// The expected ExecuteRequest.
+  ///
+  pub execute_request: bazel_protos::remote_execution::ExecuteRequest,
+  ///
+  /// Vec of Operation response for Execution or GetOperation requests. Will be returned in order.
+  ///
+  pub operation_responses: Vec<bazel_protos::operations::Operation>,
+}
+
+///
+/// A server which will answer ExecuteRequest and GetOperation gRPC requests with pre-canned responses.
+///
+pub struct TestServer {
```

Comment:
Based on naming, `TestServer` and `MockServer` seemed like peers in the ownership hierarchy, rather than one struct that holds another... changing one or the other to differentiate them would be good.

Strawman, should `MockServer` be named something like `MockResponder`?

Reply:
Done